app/models/types_de_champ/prefill_repetition_type_de_champ.rb

```diff
@@ -0,0 +1,69 @@
+class TypesDeChamp::PrefillRepetitionTypeDeChamp < TypesDeChamp::PrefillTypeDeChamp
+  include ActionView::Helpers::UrlHelper
+  include ApplicationHelper
+
+  def possible_values
+    [
+      I18n.t("views.prefill_descriptions.edit.possible_values.#{type_champ}_html"),
+      subchamps_all_possible_values
+    ].join("</br>").html_safe # rubocop:disable Rails/OutputSafety
+  end
+
+  def example_value
+    [row_values_format, row_values_format]
+  end
+
+  def to_assignable_attributes(champ, value)
+    return [] unless value.is_a?(Array)
+
+    value.map.with_index do |repetition, index|
+      PrefillRepetitionRow.new(champ, repetition, index, @revision).to_assignable_attributes
+    end.reject(&:blank?)
+  end
+
+  private
+
+  def subchamps_all_possible_values
+    "<ul>" + prefillable_subchamps.map do |prefill_type_de_champ|
+      "<li>champ_#{prefill_type_de_champ.to_typed_id_for_query}: #{prefill_type_de_champ.possible_values}</li>"
+    end.join + "</ul>"
+  end
+
+  def row_values_format
+    @row_example_value ||=
+      prefillable_subchamps.map do |prefill_type_de_champ|
+      ["champ_#{prefill_type_de_champ.to_typed_id_for_query}", prefill_type_de_champ.example_value.to_s]
+    end.to_h
+  end
+
+  def prefillable_subchamps
+    @prefillable_subchamps ||=
+      TypesDeChamp::PrefillTypeDeChamp.wrap(@revision.children_of(self).filter(&:prefillable?), @revision)
+  end
+
+  class PrefillRepetitionRow
+    attr_reader :champ, :repetition, :index, :revision
+
+    def initialize(champ, repetition, index, revision)
+      @champ = champ
+      @repetition = repetition
+      @index = index
+      @revision = revision
+    end
+
+    def to_assignable_attributes
+      return unless repetition.is_a?(Hash)
+
+      row = champ.rows[index] || champ.add_row(champ.dossier_revision)
+
+      repetition.map do |key, value|
+        next unless key.is_a?(String) && key.starts_with?("champ_")
+
+        subchamp = row.find { |champ| champ.type_de_champ_to_typed_id_for_query == key.split("_").last }
```

Comment:
```suggestion
        subchamp = row.find { |champ| champ.type_de_champ_to_typed_id_for_query == key.split("_").last.gsub(/==$/, '') }
```
Comme ça l'id complet continue de fonctionner
Comme ça si quelqu'un passe l'id copié

Reply:
Dans ce cas là tu les fais fonctionner pour les sous champs de la répétition mais pas pour les champs "classiques" si je ne me trompe pas. Genre un champ texte classique ne fonctionnera toujours pas correctement. Et pour ça je ne vois pas de fix évident

Reply:
hmm j'avais l'impression qu’ailleurs on ne ferait pas de comparaison entre `typed_id` mais on passait toujours par `Champ.id_from_typed_id`. Mais il est possible que j'ai manqué un truc.

Reply:
Je vois ça :
```ruby
  def stable_id_from_typed_id(prefixed_typed_id)
    return nil unless prefixed_typed_id.starts_with?("champ_")

    Champ.id_from_typed_id(prefixed_typed_id.gsub("champ_", "")).to_i
  rescue
    nil
  end
```

ça serait bien d'utiliser le même code pour les souschamps

Reply:
Bonne remarque, je peux changer le matching pour utiliser cette méthode ! Je crois que ça ne change rien pour la problématique ci-dessus des champs hors répétition par contre non?

Reply:
Hum, cette méthode est dans les prefill_params. Va falloir l'extraire si je veux l'utiliser

Reply:
J'ai pushé le changement. Mais ça ne change pas la problématique des === je pense

Reply:
Dans les exemples on propose la variante sans les `==` mais si je vais générer un payload à partir des id récupéré dans le schéma, ça marchera quand même

Reply:
Vraiment je ne suis pas sûr de te suivre. Si pour le champ texte (pas répétition, champ texte classique), j'ai dans mon URL quelque chose du genre `champ_blabla===value`, ce sera automatiquement parsé par Rails en `{"champ_Q2hhbXAtNQ"=>"==Texte court"}`

Reply:
Hum j'attends pour merge alors! J'aimerais bien vérifier lundi proprement que ça fonctionne comme on veut dans les 2 cas dont tu parles. Et je vais bientôt couper pour ce soir !

Reply:
ok. mais je suis assez confiant que ça marche :)

Reply:
Ca marche ! Je merge :)